client/validators.py

```diff
@@ -1,9 +1,18 @@
 import re
 from django.core.exceptions import ValidationError
+from django.core.validators import RegexValidator
 
 ID_NUMBER_REGEX = r'^\d{8,9}$'
 
 
+# Checks if the phone number starts with +972/0, and then if the second part is one of:
+# 1. a land line geographic based number - 2/3/4/8/9 and afterwards 7 digits.
+# 2. a new number country prefix - 7 and afterwards 8 digits.
+# 3. a mobile number - 5 and afterwards 8 digits.
+PHONE_NUMBER_REGEX = re.compile(r"^(\+972[- ]?|0)(([23489]|[57]\d)-?\d{3}-?\d{4})$")
```

Comment:
It's kind of nit-picky but maybe switch to
```regex
^(\+972-?|0)(([23489]-?\d{7})|[5|7]\d-?(\d{4}-?\d{3}|\d{3}-?\d{4}))$
```
so 054-1234-567 and 054-123-4567 will both be allowed.

Reply:
Why don't you simply remove hyphens altogether? would greatly simplify the regex and reduce possible errors.

Reply:
Also would standarize numbers.

Reply:
Ok, this is the standard format 054-123-4567 but I don't want to confuse the users, so both ways will be allowed.
Also, I will add the option to have whitespace after 050

Reply:
@astar26 the hyphens are optional.
@ykrinsky I talked to @astar26 , you can just use the solution I gave above for the hyphens and that's alright.